src/runtime_src/core/edge/drm/zocl/zocl_bo.c

```diff
@@ -430,7 +430,7 @@ zocl_userptr_bo_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
 
 	/* Physical address must be continuous */
 	if (sg_count != 1) {
-		DRM_ERROR("User buffer is not physical contiguous\n");
```

Comment:
I agree with you. That's my original thought.
This request if from marketing guys. They feel the system is about to crash or the XRT driver is in bad status. They don't even know if it is still safe to run any application.

Reply:
I see. In that case, how about we log the message more explicitly and do not scare the user. Something like "This user buffer %p can not be used as a buffer object because it is not physically contiguous."
The problem to change it as a warning is user will think there is something not quite right but it is still functional, which is not the case.

Reply:
User would get a bad handle returned by xclAllocUserPtrBO(). They know something is wrong and it will not be functional. When they look at dmesg log, they would see this warning and realize this is the root cause.
I think we could refine this message to "User buffer %p is not physically contiguous. Not able to create buffer object handle." I would think this is a warning. It means something is not quite right, but driver is still functional.

Reply:
I am OK to use DRM_WARN here.

How every, it does not seem to me that we should not use DRM_ERROR whenever driver is functional. For example, when somebody wants to use a BO handle which does not exist, we use DRM_ERROR to log this failure.